eth/downloader/downloader.go

```diff
@@ -1352,6 +1322,11 @@ func (d *Downloader) fetchNodeData() error {
 		deliver = func(packet dataPack) (int, error) {
 			start := time.Now()
 			return d.queue.DeliverNodeData(packet.PeerId(), packet.(*statePack).states, func(err error, delivered int) {
+				// If the peer gave us nothing, stalling fast sync, drop
+				if delivered == 0 {
```

Comment:
Though I agree that prior to 1.5 we should probably walk over our "stall" scenarios are reevaluate them in light of the current network stats. As we're having much poorer peers, it might be worthwhile to see where we can be a bit more lax and where we still need to be strict.